java/arcs/core/analysis/EntityTypeUnion.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.analysis
+
+import arcs.core.data.EntityType
+import arcs.core.data.FieldName
+import arcs.core.data.FieldType
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+
+/**
+ * Computes the union of the two [Schema] instances. Returns [Outcome.Failure] if the union
+ * is not possible as the inputs are incompatible.
+ */
+fun Schema.union(other: Schema): Outcome<Schema> {
```

Comment:
This would be a great candidate for using `infix fun`. Then you can call it like so: `schema1 union schema2`

Reply:
Good idea. Done.